db/schema_tables.hh

```diff
@@ -214,10 +240,21 @@ future<lw_shared_ptr<query::result_set>> extract_scylla_specific_keyspace_info(d
 
 std::vector<mutation> make_create_type_mutations(lw_shared_ptr<keyspace_metadata> keyspace, user_type type, api::timestamp_type timestamp);
 
+// Given a set of rows that is sorted by keyspace, create types for each keyspace.
+// The topological sort in each keyspace is necessary when creating types, since we can only create a type when the
+// types it reference have already been created.
+future<std::vector<user_type>> create_types(replica::database& db, const std::vector<const query::result_set_row*>& rows);
+
 future<std::vector<user_type>> create_types_from_schema_partition(keyspace_metadata& ks, lw_shared_ptr<query::result_set> result);
 
+std::vector<data_type> read_arg_types(replica::database& db, const query::result_set_row& row, const sstring& keyspace);
+
+future<shared_ptr<cql3::functions::user_function>> create_func(replica::database& db, const query::result_set_row& row);
+
 seastar::future<std::vector<shared_ptr<cql3::functions::user_function>>> create_functions_from_schema_partition(replica::database& db, lw_shared_ptr<query::result_set> result);
```

Comment:
Perhaps these functions should migrate to data_dictionary/ ?

Reply:
@avikivity what do you think?

Reply:
data_dictionary is a read-only view of the schema. It reflects what's stored, it doesn't store itself.

This way it can be used for mocking (and not have everything tested in a full-system test)